blockchain/blockindex.go

```diff
@@ -260,6 +260,16 @@ func (node *blockNode) RelativeAncestorCtx(distance int32) HeaderCtx {
 	return ancestor
 }
 
+// IsAncestor returns if the other node is an ancestor of this block node.
+func (node *blockNode) IsAncestor(otherNode *blockNode) bool {
+	ancestor := node.Ancestor(otherNode.height)
+	if ancestor == node {
```

Comment:
When would this case ever be triggered (`ancestor == node`)? 

From revisiting the implementation of `Ancestor`, only if `n` is `nil` would the same node be returned, or if they have an equal height. Perhaps we can make the latter case more explicit by just doing a height comparison?

Reply:
Yeah the only case this would be triggered is if the equal height is passed to `Ancestor`. I'll change it to a height comparison

Reply:
Addressed in the latest push